pkg/components/lock/lock_config.go

```diff
@@ -17,24 +18,32 @@ const (
 	separator = "||"
 )
 
-var lockConfiguration = map[string]*StoreConfiguration{}
+var (
+	locksConfigurationMu sync.RWMutex
+	lockConfiguration    = map[string]*StoreConfiguration{}
+)
 
 type StoreConfiguration struct {
 	keyPrefixStrategy string
 }
 
 func SaveLockConfiguration(storeName string, metadata map[string]string) error {
-	strategy := strings.ToLower(metadata[strategyKey])
-	if strategy == "" {
-		strategy = strategyDefault
-	} else {
-		err := checkKeyIllegal(metadata[strategyKey])
-		if err != nil {
-			return err
+	strategy := strategyDefault
+	for k, v := range metadata {
+		if strings.ToLower(k) == strategyKey { //nolint:gocritic
```

Comment:
any reason not to use `EqualFold`?

Reply:
They're both guaranteed to be "simple" ASCII strings, so `EqualFold` is not needed (because it considers Unicode runes, it's more complex and slow, and unnecessary here; see docs: https://pkg.go.dev/strings#EqualFold )